habitat-lab/habitat/core/env.py

```diff
@@ -128,6 +128,28 @@ def __init__(
             }
         )
         self.action_space = self._task.action_space
+
+        # Sharing controller between actions
+        # This should go somewhere else, but it is unclear where
+        # we want to share a controller across actions since the controller
+        # is stateful
+        from habitat_baselines.articulated_agent_controllers import (
```

Comment:
Absolutely not. habitat-baselines cannot be a dependency of habitat-lab.

Reply:
Totally agreed. I need actions to share the humanoid controller and I was not sure the best place to set them. I am looking at whether this can be added in actions controller. Otherwise, this is a good argument for moving the humanoid controllers into Habitat-Lab, as suggested in the previous PR.

Reply:
Addressed in latest commit. There is a separate PR for that here https://github.com/facebookresearch/habitat-lab/pull/1202